v2/api/containerservice/customizations/managed_clusters_agent_pool_extensions.go

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package customizations
+
+import (
+	"context"
+	"fmt"
+
+	containerservice "github.com/Azure/azure-service-operator/v2/api/containerservice/v1beta20210501storage"
+	"github.com/Azure/azure-service-operator/v2/internal/genericarmclient"
+	"github.com/Azure/azure-service-operator/v2/internal/set"
+	"github.com/Azure/azure-service-operator/v2/internal/util/kubeclient"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/extensions"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+)
+
+var _ extensions.PreReconciliationChecker = &ManagedClustersAgentPoolExtension{}
+
+// If an agent pool has a provisioningState in this set, it will reject any attempt to PUT a new state out of hand;
+// so there's no point in even trying. This is true even if the PUT we're doing will have no effect on the state of the
+// cluster.
+var blockingManagedClustersAgentPoolProvisioningStates = set.Make(
```

Comment:
consider making states lowercase and below we can use `toLower` while matching states for consistency?